### ISSUE TITLE
Python binary in conda behaves buggy when using pyside2

### ISSUE THREAD

**Issue by rggjan** [ContinuumIO/anaconda-issues#11297]
I'm using pyside2 (Qt for python) in an anaconda environment. However, there is some kind of bug in the UI, where updates are not correctly executed.

When using pyenv, everything works fine, but using conda (with the exact same python version and pyside2 pip) there is this update bug. Copying the pyenv python3.6.8 binary to conda fixes the issue.

See also https://bugreports.qt.io/browse/PYSIDE-1083

I wonder: What is the difference from the conda binary to the pyside2 python binary? There has to be some difference which introduces this issue...

I can reproducing when downloading and installing miniconda, creating a python=3.6.8 environment and pip installing pyside2.

##### Anaconda or Miniconda version:

python                    3.6.8                haf84260_0
pyside2                   5.13.1                   pypi_0    pypi

##### Operating System:

OSX 10.15 Beta

##### `conda info`

```
     active environment : foo
    active env location : /Users/user/miniconda/envs/foo
            shell level : 1
       user config file : /Users/user/.condarc
 populated config files : /Users/user/.condarc
          conda version : 4.7.12
    conda-build version : not installed
         python version : 3.6.8.final.0
       virtual packages :
       base environment : /Users/user/miniconda  (writable)
           channel URLs : https://repo.anaconda.com/pkgs/main/osx-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/osx-64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /Users/user/miniconda/pkgs
                          /Users/user/.conda/pkgs
       envs directories : /Users/user/miniconda/envs
                          /Users/user/.conda/envs
               platform : osx-64
             user-agent : conda/4.7.12 requests/2.22.0 CPython/3.6.8 Darwin/19.0.0 OSX/10.15
                UID:GID : 501:20
             netrc file : None
           offline mode : False
```

</details>

**Comment by mingwandroid** [comment]
The python extension modules built by the Qt developers will incorrectly link to a python DSO. That would be my guess.

**Comment by rggjan** [comment]
How to check if this is really the issue? And if this is the issue, how could it be fixed?

**Comment by patricksnape** [comment]
@mingwandroid I also just tried this out in conda-forge and I'm seeing the bug also manifest there. Just wondering if we can nail this down with respect to what is causing the issue.

I'm a bit confused why using pyenv and pip installing the pyside2 package works but using either vanilla conda or conda-forge does not?

**Comment by patricksnape** [comment]
Yes absolutely sorry, thanks for looking at all! As far as I can see the pyside wheel actually bundles its own copy of QT so assuming I don't have QT at all in the environment I wouldn't expect it to be a conflict between conda QT and pyside QT. This issue is manifesting for me on OSX but I haven't test Linux.

```
conda create -n pyside-test python=3.6
pip install pyside
```

```
import sys

from PySide2.QtCore import QSize, Qt
from PySide2.QtWidgets import *


class GraphicsView(QGraphicsView):
    def sizeHint(self):
        # Return scene size by default
        return QSize(500, 500)

    def __init__(self):
        super().__init__()
        self.setScene(QGraphicsScene())
        self.scene().addText('ABC')
        self.scale(20, 20)

    def test(self):
        self.scale(0.9, 0.9)
        # self.update() <= This doesn't help either


class Test(QWidget):
    def __init__(self):
        super().__init__()

        layout = QHBoxLayout()
        self.setLayout(layout)

        view = GraphicsView()
        layout.addWidget(view)

        auto_button = QPushButton('auto')
        auto_button.clicked.connect(view.test, Qt.AutoConnection) # Also doesn't work with DirectConnection
        layout.addWidget(auto_button)

        queued_button = QPushButton('queued')
        queued_button.clicked.connect(view.test, Qt.QueuedConnection)
        layout.addWidget(queued_button)


app = QApplication(sys.argv)
view = Test()
view.show()
sys.exit(app.exec_())
```

In the app, the left button doesn't work but the right one does. "working" means that the text decreases in size.

Note this issue seems to manifest also when using the conda-forge python interpreter and pyside2 installed from conda-forge rather than pip.

However, it seems this works:

```
/Users/psnape/.pyenv/versions/3.6.9/bin/python test.py
```

where the python version was freshly installed using

```
brew install pyenv
pyenv install 3.6.9
/Users/psnape/.pyenv/versions/3.6.9/bin/pip install pyside2
```

**Comment by mingwandroid** [comment]
Once you install anything using dynamically linked C++ from pip into a conda env, the libc++ loaded is essentially random and bad things will happen, at least that's the case on Linux, on macOS it's probably worse! 😕

Can you file a bug against conda-forge's pyside2 though? Your test case would be handy. 

This is, to be honest, one of condas big advantages. Cohesion.

**Comment by patricksnape** [comment]
@mingwandroid Thanks for the help -> opened appropriate issue and probably we should close this and move discussion there as I agree with @mingwandroid about the futility of trying to mix wheels with conda and expecting the correct behaviour.

**Comment by rggjan** [comment]
Another interesting observation: Running the example with dark mode enabled works fine with pyside2 but ignores the dark mode setting when using the anaconda compiled python2.

<img width="813" alt="Screenshot 2019-10-22 at 10 35 23" src="https://user-images.githubusercontent.com/128554/67270904-d7504580-f4b9-11e9-8e01-acd6adc570b6.png">
<img width="813" alt="Screenshot 2019-10-22 at 10 35 37" src="https://user-images.githubusercontent.com/128554/67270905-d7504580-f4b9-11e9-849d-63fdc3591e82.png">

I suspect the reason for both issues is that anaconda compiles using a very old version of clang (4.0 instead of 11.0) and links against a very old OSX SDK (10.9), and both are not using Xcode to do the compiling:

https://docs.conda.io/projects/conda-build/en/latest/resources/compiler-tools.html

Qt builds using a newer deployment target:

https://github.com/conda-forge/conda-forge.github.io/issues/891

https://github.com/conda-forge/qt-feedstock/blob/master/recipe/conda_build_config.yaml

 

So I suspect that some OSX features are "enabled" in the Qt libs, but "disabled" in the python binary as it links against an old SDK, or something like this. You can see this difference also in the python binaries:

```
➜ otool -L ~/.pyenv/versions/3.6.8/bin/python3.6
➜ otool -L ~/.pyenv/versions/3.6.8/bin/python3.6/Users/user/.pyenv/versions/3.6.8/bin/python3.6:
/System/Library/Frameworks/CoreFoundation.framework/Versions/A/CoreFoundation (compatibility version 150.0.0, current version 1665.10.0)
/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1281.0.0)
```
 vs

```
➜ otool -L ~/miniconda/envs/foo/bin/python3.6
/Users/user/miniconda/envs/foo/bin/python3.6:/Users/user/miniconda/envs/foo/bin/python3.6:
/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1197.1.1)
```

**Comment by rggjan** [comment]
Next observation: I tried rebuilding different python versions from

https://github.com/conda-forge/python-feedstock 

I rebuilt python 3.6.7 with this basic conda_build_config.yaml:

```
CONDA_BUILD_SYSROOT:
 - optMacOSX10.9.sdk
DEBUG_C:
 - yes
```

And had (as expected) the same issues as with the conda or conda-forge built python. However, adding:
```
c_compiler_version:
 - 9
```
(and making sure conda-forge was in the channels, as conda-forge provides clang-9)

fixes the original issue with the updates, although the dark mode is still not working. I suspect changing the OSX SDK or minimum required version would fix this as well...

So there seems to be some compatibility issue regarding the clang version Qt / PySide2 and Python is compiled with...

**Comment by rggjan** [comment]
Next observation: I tried rebuilding different python versions from

https://github.com/conda-forge/python-feedstock

I rebuilt python 3.6.7 with this basic conda_build_config.yaml:

 ```
CONDA_BUILD_SYSROOT:
 - optMacOSX10.9.sdk
DEBUG_C:
 - yes
```
And had (as expected) the same issues as with the conda or conda-forge built python. However, adding:

```
c_compiler_version:
 - 9
```
(and making sure conda-forge was in the channels, as conda-forge provides clang-9)

fixes the original issue with the updates, although the dark mode is still not working. I suspect changing the OSX SDK or minimum required version would fix this as well...

So there seems to be some compatibility issue regarding the clang version Qt / PySide2 and Python is compiled with...
